aten/src/ATen/native/Indexing.cpp

```diff
@@ -69,27 +70,24 @@ static std::vector<Tensor> expandByteTensors(const Tensor & self, TensorList ind
       }
       // Replace with nonzeros
       auto nonzero = index.nonzero();
-      auto special_empty = false;
       for (int64_t j = 0; j < index.dim(); j++) {
-        if (special_empty) {
-          // We can't call select on an empty tensor so we just create an empty
-          // tensor.
-          result.emplace_back(nonzero.type().tensor());
-        } else {
-          result.emplace_back(nonzero.select(1, j));
-        }
+        result.emplace_back(nonzero.select(1, j), false);
       }
     } else {
-      result.emplace_back(index);
+      result.emplace_back(index, true);
     }
   }
   return result;
 }
 
-static bool hasContiguousSubspace(TensorList tl) {
+static bool hasContiguousSubspace(std::vector<std::tuple<Tensor, bool>> tl) {
```

Comment:
This should be `const std::vector<std::tuple<Tensor, bool>>&` or `at::ArrayRef<std::tuple<Tensor, bool>>`, right now you're making a copy each time.